api/src/main/java/org/openmrs/ui/framework/DefaultFailedAuthenticationHandler.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *  
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.ui.framework;
+
+import java.util.List;
+
+import org.openmrs.ui.framework.page.PageRequest;
+
+public class DefaultFailedAuthenticationHandler implements FailedAuthenticationHandler {
+	
+	public String handle(PageRequest pageRequest, List<String> requiredPrivileges) {
+		return getRedirectUrl();
+	}
+	
+	public String getRedirectUrl() {
+		//This is the only login page we can guarantee exists for now,
+		//but how about after removing the legacy UI from the platform?
+		return "/login.htm";
```

Comment:
Another downside with this default handler being in this module is that it has to default to sending the user to a login page. If they don't have the privilege and they are already logged in, they get redirect to the home page e.g this is the how it is done by the logic in the reference application module. It would be nice to display some error message that says they are required to have certain privileges but we can't do that here because the uiframework module don't know about reporting error messages in the header of pages. Alternatively we could log them out and take them to the login page or we could take them to the error page in the reference uiframework module if they are already logged in and ask them to login with a different account